### PR TITLE
fix(api): fixed FF engined-based module loading failing in analysis

### PR DESCRIPTION
**Pull request by jbleon95**
# Overview
Followup to PR #11619, part of RCORE-257.

Fixed issue where engine-based module loading was not handling simulating cases.

# Changelog
- Module Loading in engine-based PAPIv2 fixed

# Review requests
Test loading module on hardware, both analysis and protocol run.

# Risk assessment
Low, fixes bug in previous PR.

### CODE REVIEW

File: api/src/opentrons/protocol_api/core/engine/protocol.py [hunk 1]
```diff
@@ -139,6 +140,22 @@ def load_labware(
...
+    def _resolve_module_hardware(
+        self, serial_number: str, model: ModuleModel
+    ) -> AbstractModule:
+        """Resolve a module serial number to module hardware API."""
+        selected_hardware: AbstractModule
+        for module_hardware in self._sync_hardware.attached_modules:
+            if serial_number == module_hardware.device_info["serial"]:
+                selected_hardware = module_hardware
+                break
+        else:
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api/core/engine/protocol.py]: is there an indentation problem here?

**Review by TamarZanzouri — APPROVED** [review]
Changes look good to me! just had one question about the indentation

File: api/src/opentrons/protocol_api/core/engine/protocol.py (hunk 1 above)
Comment (jbleon95) [reply api/src/opentrons/protocol_api/core/engine/protocol.py]: No, I'm using a for/else loop clause here, the else statement will only be entered if the for loop does not `break`

Comment (mcous) [reply api/src/opentrons/protocol_api/core/engine/protocol.py]: `for/else` is a recent learning for me, and I gotta say, it's not super easy for me to read. Especially with this split to its own method, I'd softly prefer a early return for readability, if it's not too much trouble

```python
def _resolve_module_hardware(
        self, serial_number: str, model: ModuleModel
    ) -> AbstractModule:
        """Resolve a module serial number to module hardware API."""
        if self.is_simulating():
            return self._sync_hardware.create_simulating_module(model)

        for module_hardware in self._sync_hardware.attached_modules:
            if serial_number == module_hardware.device_info["serial"]:
                return module_hardware

        raise RuntimeError(f"Could not find specified module: {model.value}")
```

Comment (TamarZanzouri) [reply api/src/opentrons/protocol_api/core/engine/protocol.py]: only in python :-) thank you for the explanation!

**Review by mcous — APPROVED** [review]
This fix will do the trick, but it won't be sufficient in the long term due to how this will produce a mismatch in reported hardware state and engine state (where there is no simulating module).

But that's a (much more complicated) problem for another day and another ticket! 🚢

**Pull request merged**